instant-xml-macros/src/de.rs

```diff
@@ -1,11 +1,54 @@
 use proc_macro2::{Ident, Span, TokenStream};
 use quote::{quote, ToTokens};
 
-use super::{discard_lifetimes, ContainerMeta, FieldMeta, Namespace};
+use super::{discard_lifetimes, ContainerMeta, FieldMeta, Namespace, VariantMeta};
 
 pub(crate) fn from_xml(input: &syn::DeriveInput) -> TokenStream {
     let ident = &input.ident;
     let container_meta = ContainerMeta::from_derive(input);
+    match &input.data {
+        syn::Data::Struct(ref data) => deserialize_struct(input, data, container_meta, ident),
+        syn::Data::Enum(_) if !container_meta.scalar => {
+            panic!("non-scalar enums are currently unsupported!")
+        }
+        syn::Data::Enum(ref data) => deserialize_enum(input, data),
+        _ => todo!(),
+    }
+}
+
+#[rustfmt::skip]
```

Comment:
Why do we need the `rustfmt::skip` here?

Reply:
Generally rustfmt misbehaves on code inside macros, so anything with a macro invocation where I want to preserve formatting, I'll use `rustfmt::skip`.